src/Interface/Character/Parses.js

```diff
@@ -73,8 +73,9 @@ class Parses extends React.Component {
     this.updateZoneMetricBoss = this.updateZoneMetricBoss.bind(this);
   }
 
-  componentDidMount() {
+  async componentDidMount() {
     this.fetchBattleNetInfo();
+    this.realmSlug = await this.getRealmSlug();
```

Comment:
This is basically the same as state, but looking at it again I don't think you can solve it without adding any kind of state. How about setting the realm to the value of `findRealm` in the `setState` on line 320? Then everything updates in the state at once.